isabelle/Doc/Specification/Guarantees.thy

```diff
@@ -14,132 +14,187 @@ begin
+without inputs with a validity interval that starts after the max time will always succeed 
 \<close>
 
-text \<open>@{thm [display,names_short, margin=40] timeOutTransaction_does_not_fail}\<close>
+text \<open>\<^bold>\<open>corollary\<close> \<^emph>\<open>timeOutTransaction\_does\_not\_fail\<close>: @{thm [display,names_short, margin=60] timeOutTransaction_does_not_fail}\<close>
 
-text \<open>Moreover, that transaction empties the account and produces payments for any lingering assets\<close>
+text \<open>Moreover, we can use the function \<^emph>\<open>isClosedAndEmpty\<close> to
+assert that the previous transaction will result in a Close contract with empty accounts.\<close>
 
-text \<open>@{thm [display,names_short, margin=40] timedOutTransaction_preserves_assets}\<close>
+text \<open>
+\<^item> isClosedAndEmpty :: @{typeof isClosedAndEmpty}\<close>
 
-section \<open>Positive Accounts\<close>
+text \<open>\<^bold>\<open>theorem\<close> \<^emph>\<open>timedOutTransaction\_closes\_contract\<close>:@{thm [display,names_short, margin=60] timedOutTransaction_closes_contract}\<close>
 
-text \<open>
+text \<open>The helper and lemmas are located in \<^emph>\<open>Timeout.thy\<close>.\<close>
+
+
+subsection \<open>Transaction Bound\label{sec:transaction-bound}\<close>
 
-There are some values for State that are allowed by its type but make no sense, especially in the
-case of Isabelle semantics where we use lists of key values to represent maps:
-\begin{enumerate}
-\item The lists represent maps, so they should have no repeated keys (@{term valid_map}).
-\item Two equal maps should be represented equally, so we force keys to be in ascending order  (@{term valid_map}).
-\item Only the accounts that contain a positive amount are relevant (@{term allAccountsPositiveState}).
-\end{enumerate}
+text \<open>There is a maximum number of transactions that can be accepted by a contract.\<close>
+
+text \<open>
+\<^item> maxTransactionsInitialState :: @{typeof maxTransactionsInitialState}
 \<close>
 
-text \<open>@{code_stmts validAndPositive_state constant: validAndPositive_state valid_state valid_map allAccountsPositive allAccountsPositiveState (Haskell)}\<close>
+text \<open>We can prove that if we run a successful trace, the number of transactions won't
+exceed this value\<close>
+
+(* should we have a maxTransactions :: Contract \<Rightarrow> Int in the semantics? *)
 
-text \<open>If the accounts are valid and possitive, then applying an input preserves that property.\<close>
+text \<open>\<^bold>\<open>lemma\<close> \<^emph>\<open>playTrace\_only\_accepts\_maxTransactionsInitialState\<close>: @{thm  [display,names_short, margin=60] playTrace_only_accepts_maxTransactionsInitialState}\<close>
 
-text \<open>@{thm [display,names_short, margin=40] playTraceAux_preserves_validAndPositive_state}\<close>
+text \<open>The helper and lemma are located in \<^emph>\<open>TransactionBound.thy\<close>.\<close>
```

Comment:
```suggestion
text \<open>The helper function and lemma are located in \<^emph>\<open>TransactionBound.thy\<close>.\<close>
```
Same. I am hoping you didn't do it because of formatting, if so you can reject the suggestion